### PR TITLE
feat(module:descriptions): add component

### LINKED ISSUE #2847: new component: descriptions
~Ref https://github.com/ant-design/ant-design/pull/14529.~
Ref https://github.com/ant-design/ant-design/pull/14645.

### PR DESCRIPTION
**Pull request by wendzhue**
close #2847

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #2847

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by wendzhue** [comment]
Should be merged after this component is shipped in antd.

**Comment by wendzhue** [comment]
I fixed the test for responsive in the grid component.

File: components/descriptions/doc/index.zh-CN.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+---
+
+成组显示多个只读字段。
+
+## 何时使用
+
+常见于详情页的信息展示。
+
+## API
+
```
Comment (hsuanxyz) [inline components/descriptions/doc/index.zh-CN.md]: 二级入口引入

File: components/descriptions/nz-descriptions-item.component.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+import { ChangeDetectionStrategy, Component, Input, TemplateRef, ViewChild, ViewEncapsulation } from '@angular/core';
+
+import { InputNumber } from 'ng-zorro-antd/core';
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  selector: 'nz-descriptions-item',
```
Comment (hsuanxyz) [inline components/descriptions/nz-descriptions-item.component.ts]:
```ts
exportAs: 'nzDescriptionsItem'
```

File: components/descriptions/nz-descriptions.component.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+  md: 3,
+  sm: 2,
+  xs: 1
+};
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  selector: 'nz-descriptions',
+  templateUrl: './nz-descriptions.component.html',
```
Comment (hsuanxyz) [inline components/descriptions/nz-descriptions.component.ts]:
```ts
exportAs: 'nzDescriptions'
```

File: components/descriptions/doc/index.zh-CN.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+---
+category: Components
+type: 数据录入
```
Comment (hsuanxyz) [inline components/descriptions/doc/index.zh-CN.md]:
```suggestion
type: 数据展示
```

File: components/descriptions/doc/index.en-US.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+---
+category: Components
+type: Data Display
+title: Descriptions
```
Comment (hsuanxyz) [inline components/descriptions/doc/index.en-US.md]:
```suggestion
title: Description List
```

File: components/core/responsive/break-point.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+export type BreakpointMap = { [index in keyof typeof Breakpoint]: string };
+
+export const responsiveMap: BreakpointMap = {
+  xs: '(max-width: 575px)',
+  sm: '(min-width: 576px)',
+  md: '(min-width: 768px)',
+  lg: '(min-width: 992px)',
+  xl: '(min-width: 1200px)',
+  xxl: '(min-width: 1600px)'
+};
```
Comment (hsuanxyz) [inline components/core/responsive/break-point.ts]: 如果 Breakpoint 类型想要 string 的话应该这样
```
export enum Breakpoint {
  xxl = 'xxl',
  xl = 'xl',
  lg = 'lg',
  md = 'md',
  sm = 'sm',
  xs = 'xs'
}
```

否则 `responsiveMap` 的 key 应该是 number

File: components/descriptions/doc/index.en-US.md (hunk 5 above)
Comment (wendzhue) [reply components/descriptions/doc/index.en-US.md]: 这个和 react 那边一致，就叫 descriptions

Comment (wendzhue) [reply components/descriptions/doc/index.en-US.md]: 这个应该是他们那边搞错了，中英文不一致

File: components/descriptions/nz-descriptions.component.html [hunk 7]
```diff
@@ -0,0 +1,34 @@
+<div *ngIf="nzTitle"
+     class="ant-descriptions-title">
+  <ng-container *nzStringTemplateOutlet="nzTitle">{{ nzTitle }}</ng-container>
+</div>
+<div class="ant-descriptions-view">
+  <table>
+    <tbody>
+      <tr class="ant-descriptions-row"
+          *ngFor="let row of itemMatrix; let i = index"
+          [attr.key]="i">
```
Comment (hsuanxyz) [inline components/descriptions/nz-descriptions.component.html]:
```suggestion
          >
```

File: components/descriptions/nz-descriptions.component.ts [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+      ),
+      this.resize$
+    )
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(() => {
+        this.prepareMatrix();
+        this.cdr.markForCheck();
+      });
+
+    this.ngZone.runOutsideAngular(() => {
```
Comment (hsuanxyz) [inline components/descriptions/nz-descriptions.component.ts]:
```suggestion
if (this.platform.isBrowser) {
    this.ngZone.runOutsideAngular(() => {
```

File: components/descriptions/doc/index.en-US.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+cols: 1
+---
+
+Empty state placeholder.
```
Comment (simplejason) [inline components/descriptions/doc/index.en-US.md]: need to sync with ant-design again

**Pull request merged**